bin/metaBackend.py

```diff
@@ -21,47 +21,506 @@
+         Connection parameters. It defaults to the value specified via -a option.
+         Use it if the database is on a different server than metaserv.
+
+    --------------------------------------------------------------------------------
+
+    ADD DB <dbName> <level> <dataRel> <owner> <accessibility> <project>
+           <mysqlAuthFile>;
+
+    It adds a database without any additional schema description. All the rest is
+    the same as for ADD DBDESCR.
+
+    Note: this is not implemented. See DM-2301
+
+    --------------------------------------------------------------------------------
+
+    ADD INSTITUTION <name>;
+
+    --------------------------------------------------------------------------------
+
+    ADD PROJECT <name>;
+
+    --------------------------------------------------------------------------------
+
+    ADD USER <mysqlUserName> <firstName> <lastName> <affilliation> <email>;
+
+    Adds a user. In the future this will create mysql user account
+    (e.g., on lsst10). <Affilliation> should be set to an existing <instShortName>
+    (these can be added using "ADD INSTITUTION" command.)
+
+    --------------------------------------------------------------------------------
+
+    QUIT;
+
+    --------------------------------------------------------------------------------
+
+    EXIT;
+
+    --------------------------------------------------------------------------------
+
+    ...more coming soon
+
+"""
+
+    def receiveCommands(self):
+        """
+        Receive user commands. End of command is determined by ';'. Multiple
+        commands per line are allowed. Multi-line commands are allowed. To
+        terminate: CTRL-D, or 'exit;' or 'quit;'.
+        """
+        line = ''
+        cmd = ''
+        prompt = "metab > "
+        while True:
+            line = raw_input(prompt).decode("utf-8").strip()
+            cmd += line + ' '
+            prompt = "metab > " if line.endswith(';') else "~ "
+            while re.search(';', cmd):
+                pos = cmd.index(';')
+                try:
+                    self._parse(cmd[:pos])
+                except (MetaBException) as e:
+                    log.error(e.__str__())
+                    print "ERROR: ", e.__str__()
+                cmd = cmd[pos+1:]
+
+    def _parse(self, cmd):
+        """
+        Parse, and dispatch to subparsers based on first word. Raise exceptions on
+        errors.
+        """
+        cmd = cmd.strip()
+        # ignore empty commands, these can be generated by typing ;;
+        if len(cmd) == 0: return
+        tokens = cmd.split()
+        tokens = [t.encode('utf8') for t in tokens]
+        t = tokens[0].upper()
+        if t in self._funcMap:
+            self._funcMap[t](tokens[1:])
+        else:
+            raise MetaBException(MetaBException.NOT_IMPLEMENTED, cmd)
+
+    def _parseAdd(self, tokens):
+        """
+        Subparser - handles ADD requests.
+        """
+        t = tokens[0].upper()
```

Comment:
Shouldn't be, but technically someone could type "ADD" without anything after. I added protection.